drivers/clock_control/clock_control_bl60x.c

```diff
@@ -0,0 +1,946 @@
+/*
+ * Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT bflb_bl60x_clock_controller
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/syscon.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/dt-bindings/clock/bflb_bl60x_clock.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clock_control_bl60x, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+#include <bouffalolab/bl60x/bflb_soc.h>
+#include <bouffalolab/bl60x/aon_reg.h>
+#include <bouffalolab/bl60x/glb_reg.h>
+#include <bouffalolab/bl60x/hbn_reg.h>
+#include <bouffalolab/bl60x/pds_reg.h>
+#include <bouffalolab/bl60x/l1c_reg.h>
+#include <bouffalolab/bl60x/extra_defines.h>
+
+#define CLK_SRC_IS(clk, src)                                                                       \
+	DT_SAME_NODE(DT_CLOCKS_CTLR_BY_IDX(DT_INST_CLOCKS_CTLR_BY_NAME(0, clk), 0),                \
+		     DT_INST_CLOCKS_CTLR_BY_NAME(0, src))
+
+#define CLOCK_TIMEOUT               1024
+#define RC32M_FREQ                  (32 * 1000 * 1000)
+#define EFUSE_RC32M_TRIM_OFFSET     0x0C
+#define EFUSE_RC32M_TRIM_EN_POS     19
+#define EFUSE_RC32M_TRIM_PARITY_POS 18
+#define EFUSE_RC32M_TRIM_POS        10
+#define EFUSE_RC32M_TRIM_MSK        0x3FC00
+
+enum bl60x_clkid {
+	bl60x_clkid_none = -1,
```

Comment:
`bl60x_clkid_none = -1,`
why do you need this ? I'm not sure if you use it.

Reply:
done